src/connections/api.js

```diff
@@ -0,0 +1,193 @@
+            userInfo: {
+              userName,
+              userId
+            },
+            ...rest
+          };
+
+          this.controller.currentResponse = this.#ws;
+          this.controller.ws = this.#ws;
+
+          try {
+            await onTwitchMessage(this.controller, 'ptzapi', userName, cmdMessage, tags);
+            this.#ws.send(JSON.stringify({ success: true, message: 'Command processed successfully' }));
+          } catch (error) {
+            this.#logger.error(`Error processing command: ${error.message}`);
+            this.#ws.send(JSON.stringify({ error: 'Failed to process command' }));
+          }
+
+        } catch (error) {
+          this.#logger.error(`Unexpected error handling message: ${error.message}`);
+        }
+      });
+
+      this.#ws.addEventListener('close', () => {
+        this.#logger.log('WebSocket connection to public API server closed');
+      });
+
+      this.#ws.addEventListener('error', (error) => {
+        this.#logger.error(`WebSocket error: ${error.message}`);
+      });
+    } catch (error) {
+      this.#logger.error(`Error creating WebSocket connection: ${error.message}`);
+    }
+  }
+
+  // Start the ping heartbeat
+  #startHeartbeat() {
+    this.#pingInterval = setInterval(() => {
+      if (this.#ws.readyState === WebSocket.OPEN) {
+        this.#ws.send(JSON.stringify({ type: 'ping' }));
+      }
+    }, 90000); // Send a ping every 90 seconds
+  }
+
+  // Send API command
+  async sendAPI(output) {
+    try {
+      const ws = this.controller ? this.controller.currentResponse : null;
+
+      if (!ws || ws.readyState !== WebSocket.OPEN) {
+        this.#logger.error('WebSocket connection not found or closed');
+        return;
+      }
+
+      try {
+        ws.send(JSON.stringify({ message: output }));
+        this.controller.currentResponse = null;
+      } catch (error) {
+        this.#logger.error(`Failed to send response: ${error.message}`);
+      }
+    } catch (error) {
+      this.#logger.error(`Unexpected error in sendAPI: ${error.message}`);
+    }
+  }
+
+  // Send broadcast message
+  async sendBroadcastMessage(message, type = 'frontend') {
+    try {
+      const ws = this.controller ? this.controller.ws : null;
+
+      if (!ws || ws.readyState !== WebSocket.OPEN) {
+        this.#logger.error('WebSocket connection not found or closed');
+        return;
+      }
+
+      try {
+        ws.send(JSON.stringify({ type: type, data: message }));
+      } catch (error) {
+        this.#logger.error(`Failed to send broadcast message: ${error.message}`);
+      }
+    } catch (error) {
+      this.#logger.error(`Unexpected error in sendBroadcastMessage: ${error.message}`);
+    }
+  }
+}
+
+// Global error handling
+process.on('unhandledRejection', (reason, promise) => {
+  console.error('Unhandled Rejection at:', promise, 'reason:', reason);
+});
+
+process.on('uncaughtException', (error) => {
+  console.error('Uncaught Exception:', error);
+});
```

Comment:
Not a fan of adding global stuff into a specific connection file -- this will stop any fatal error across the entire bot from killing the process

Reply:
This was same as below, to avoid having the API crash the node, but you are correct not good practice to have global stuff in the connection file, will remove.